pkg/keys/ed25519.go

```diff
@@ -0,0 +1,123 @@
+package keys
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+func init() {
+	SignerMap.Store(data.KeySchemeEd25519, NewP256Signer)
+	VerifierMap.Store(data.KeySchemeEd25519, NewP256Verifier)
+}
+
+func NewP256Signer() Signer {
+	return &ed25519Signer{}
+}
+
+func NewP256Verifier() Verifier {
+	return &ed25519Verifier{}
+}
+
+type ed25519Verifier struct {
+	PublicKey data.HexBytes `json:"public"`
+	key       *data.Key
+}
+
+func (e *ed25519Verifier) Public() string {
+	return string(e.PublicKey)
+}
+
+func (e *ed25519Verifier) Verify(msg, sig []byte) error {
+	if !ed25519.Verify([]byte(e.PublicKey), msg, sig) {
+		return ErrInvalid
+	}
+	return nil
+}
+
+func (e *ed25519Verifier) MarshalKey() *data.Key {
+	return e.key
+}
+
+func (e *ed25519Verifier) UnmarshalKey(key *data.Key) error {
+	e.key = key
+	if err := json.Unmarshal(key.Value, e); err != nil {
+		return errors.New("unmarshalling ed25519 key")
```

Comment:
In the error message, please indicate that something went wrong. Example: "tuf: signature verification **failed**". I think prefixing that with "tuf:" is also a good idea.

Reply:
Should I just return the JSON error then? 
`json: cannot unmarshal bool into Go value of type keys.ed25519Verifier"`

Ultimately this gets called through `GetVerifier` -- which will wrap the error string like `tuf: error unmarshalling key: json: cannot unmarshal..."

Reply:
Fixed up error messages to contain `tuf: ` if they are generated from logic here and updated the style to indicate failure